spec/system/check_records/user_accepts_terms_and_conditions_spec.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "Terms and conditions acceptance", host: :check_records do
+  include AuthorizationSteps
+  include AuthenticationSteps
+  include ActivateFeaturesSteps
+
+  scenario "User accepts terms and conditions", test: :with_stubbed_auth do
+    given_the_check_service_is_open
+    given_the_terms_and_conditions_feature_flag_is_enabled
+    when_i_sign_in_via_dsi
+    then_i_am_signed_in
+    and_i_am_redirected_to_the_terms_and_conditions_page
+    then_i_click_accept
+    then_i_am_taken_to_the_root
+    and_i_see_the_successful_notification
+    and_the_dsi_user_has_been_updated
```

Comment:
Once the timestamp check has also been added, this spec can be expanded to travel forward in time and assert that the user is prompted again.